crates/kubelet/src/node.rs

```diff
@@ -389,3 +383,66 @@ fn lease_spec_definition(node_name: &str) -> serde_json::Value {
         }
     )
 }
+
+/// Defines the labels that will be applied to this node
+///
+/// Default values and passed node-labels arguments are injected by config.
+fn node_labels_definition(arch: &str, config: &Config) -> HashMap<String, String> {
+    // Add mandatory static labels
+    let mut labels = HashMap::new();
+    labels.insert("beta.kubernetes.io/os".to_string(), "linux".to_string());
+    labels.insert("kubernetes.io/os".to_string(), "linux".to_string());
+    labels.insert("kubernetes.io/role".to_string(), "agent".to_string());
+    labels.insert("type".to_string(), "krustlet".to_string());
+    // add the mandatory labels that are dependent on injected values
+    labels.insert("beta.kubernetes.io/arch".to_string(), arch.to_string());
+    labels.insert("kubernetes.io/arch".to_string(), arch.to_string());
+    labels.insert(
+        "kubernetes.io/hostname".to_string(),
+        config.hostname.to_string(),
+    );
+
+    let k8s_namespace = "kubernetes.io";
+    // namespaces managed by this method - do not allow user injection
+    let managed_namespace_labels = vec![
+        "beta.kubernetes.io/arch",
+        "beta.kubernetes.io/os",
+        "kubernetes.io/arch",
+        "kubernetes.io/hostname",
+        "kubernetes.io/os",
+        "kubernetes.io/role",
+        "type",
+    ];
+    let allowed_k8s_namespace_labels = vec![
+        "beta.kubernetes.io/instance-type",
+        "failure-domain.beta.kubernetes.io/region",
+        "failure-domain.beta.kubernetes.io/zone",
+        "failure-domain.kubernetes.io/region",
+        "failure-domain.kubernetes.io/zone",
+        "kubernetes.io/instance-type",
+    ];
+
+    // Attempt to append node labels from passed arguments.
+    // First, check for managed namespace and log exclusion
+    // Next, check if label contains k8s namespace and ensure it's allowable
+    // Else, if not k8s namspace, insert
+    let user_labels = config.node_labels.clone();
+
+    for (key, value) in user_labels.iter() {
+        if managed_namespace_labels.contains(&&key[..]) {
+            warn!(
+                "User provided node label {} omitted. Namespace label managed by runtime.",
+                key
+            );
+        } else if key.contains(k8s_namespace)
+            && (key.starts_with("kubelet.kubernetes.io")
+                || key.starts_with("node.kubernetes.io")
+                || allowed_k8s_namespace_labels.contains(&&key[..]))
+        {
+            labels.insert(key.to_owned(), value.to_owned());
```

Comment:
this call can be moved after the `if` block, and the `else` clause can be removed. Net result will be the same thing, but it will remove the `clippy` error

https://github.com/rust-lang/rust-clippy

Reply:
I think clippy actually saved me from myself here.  I added a warn! for this second error condition.

Reply:
@bacongobbler kk.  I think it's ready for a review.